Updates to Readmes (#544)
Esri/arcgis-runtime-samples-java: geometry/list-transformations-by-suitability/README.md

* GeometryEngine
* TransformationCatalog

## Additional information

Some transformations aren't available until transformation data is provided.

> This sample uses a `GeographicTransformation`, which extends the `DatumTransformation` class. As of 100.9, ArcGIS Runtime also includes a `HorizontalVerticalTransformation`, which also extends `DatumTransformation`. The `HorizontalVerticalTrasnformation` class is used to transform coordinates of z-aware geometries between spatial references that have different geographic and/or vertical coordinate systems.

Reviewer: Should this be `Z-aware`? Not sure when you capitalise.

```suggestion
This sample uses a `GeographicTransformation`, which extends the `DatumTransformation` class. As of 100.9, ArcGIS Runtime also includes a `HorizontalVerticalTransformation`, which also extends `DatumTransformation`. The `HorizontalVerticalTrasnformation` class is used to transform coordinates of Z-aware geometries between spatial references that have different geographic and/or vertical coordinate systems.
```

Author: Not 100%, seems to be z-aware when I did a search on documentation, but refer to a Z when there's not a hyphen. I'll leave as is based on the documentation check.

> This sample uses a `GeographicTransformation`, which extends the `DatumTransformation` class. As of 100.9, ArcGIS Runtime also includes a `HorizontalVerticalTransformation`, which also extends `DatumTransformation`. The `HorizontalVerticalTrasnformation` class is used to transform coordinates of z-aware geometries between spatial references that have different geographic and/or vertical coordinate systems.

Reviewer 2: It looks like in our sample we don't use the `GeographicTransformation` class: just the `DatumTransformation` class. Whenever I do a ReadMe update which contains new classes, I always do a quick "Find..." within the sample just to sanity check if we use the class from the generic sample description or not: samples can vary slightly between SDKs and also in name, so it's a good habit to get into! 

What is the purpose of the addition of this additional information here: is it intended that this sample use the `HorizontalVerticalTransformation`? Does the code require updating? If not then we can just reword to avoid specifically saying that our samples uses the `GeographicTransformation`: but you should probably double check the wording with the author of the common sample readme to be sure this still delivers the message intended. 

```suggestion
As of 100.9, ArcGIS Runtime also includes a `HorizontalVerticalTransformation`, which extends `DatumTransformation`. The `HorizontalVerticalTrasnformation` class is used to transform coordinates of z-aware geometries between spatial references that have different geographic and/or vertical coordinate systems.
```

Author: Checked these queries on CS Issue. And amended as required.

This sample can be used with or without provisioning projection engine data to your device. If you do not provision data, a limited number of transformations will be available.

To download projection engine data to your device:

1. Login to the [ArcGIS for Developers site](https://developers.arcgis.com/sign-in/) using your Developer account.
2. In the Dashboard page, click '[Download APIs and SDKs](https://developers.arcgis.com/downloads/data)' and go to the `Supplemental ArcGIS Runtime Data` tab.
3. Click the download button next to `Projection Engine Data` to download projection engine data to your computer.
4. Unzip the downloaded data on your computer.

> 5. Create an `~/ArcGIS/Runtime/Data/PEDataRuntime` directory on your device and copy the files to this directory.

Reviewer 2: it's possible we need to refine this further for Java: having a look at our `TransformationCatalog` class, there is guidance there for where the default location for such projection info is:

https://developers.arcgis.com/java/latest/api-reference/reference/com/esri/arcgisruntime/geometry/TransformationCatalog.html#getProjectionEngineDirectory()

Note that it is also possible to set the directory location:
https://developers.arcgis.com/java/latest/api-reference/reference/com/esri/arcgisruntime/geometry/TransformationCatalog.html#setProjectionEngineDirectory(java.lang.String)

in which case we'd need to say something like the following, but I'd like a second opinion from @colinanderson here first please to sanity check what I'm saying! 

```suggestion
5. Find the `<application directory>/resources/pedata` directory on your device and copy the files to this directory.
```

Author: As discussed with Rachael/Mark, the data in that directory is downloaded with the SDK for Java so not needed in the readme
